config.go

```diff
@@ -479,6 +481,24 @@ func loadConfig() (*config, []string, error) {
 		return nil, nil, err
 	}
 
+	// Set the default policy for relaying non-standard transactions
+	// according to the default of the active network. The set
+	// configuration value takes precedence over the default value for the
+	// selected network.
+	switch {
+	case cfg.RelayNonStd && cfg.RejectNonStd:
+		str := "%s: rejectnonstd and relaynonstd cannot be used " +
+			"together -- choose only one"
+		err := fmt.Errorf(str, funcName)
+		fmt.Fprintln(os.Stderr, err)
+		fmt.Fprintln(os.Stderr, usageMessage)
+		return nil, nil, err
+	case cfg.RejectNonStd && activeNetParams.RelayNonStdTxs:
```

Comment:
Agreed. I originally had something almost identical to that, but somehow convinced myself that the logic was incorrect (a bit of overthinking). The logic has been updated the clearer fragment you posted above.